### PR TITLE
update readme + data transfer instructions

### PR DESCRIPTION
**Pull request by klemens-floege**
updated the instructions for code + data transfer

### CODE REVIEW

File: minimal_example.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+dm = ExampleDataModule()
+test_datasets = dm.test_datasets()
+
+model = xgb.XGBClassifier(
+    eval_metric="logloss", random_state=42, n_estimators=200,
+)
+
+# 🚀  Quick sanity check: classification
+scores = benchmark_datasets(model, test_datasets)
+print("Mean scores across synthetic datasets:")
```
Comment (noahho) [inline minimal_example.py:24]: synthetic datasets -> example datasets

Comment (noahho) [inline minimal_example.py:24]:
```suggestion
print("Mean scores across example datasets:")
```

File: minimal_example.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# 🚀  Quick sanity check: regression
+dm = ExampleDataModule()
+test_datasets = dm.test_datasets()
+
+model = xgb.XGBRFRegressor(
+    eval_metric="logloss", random_state=42, n_estimators=200,
+)
+
+scores = benchmark_datasets(model, test_datasets)
+print("Mean scores across synthetic datasets:")
```
Comment (noahho) [inline minimal_example.py:38]:
```suggestion
print("Mean scores across example datasets:")
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "ds_requirement_guide"
+version = "0.1.0"
+description = "Data Science Requirements Guide"
+readme = "README.md"
+requires-python = ">=3.8"
+license = {text = "MIT"}
```
Comment (noahho) [inline pyproject.toml:11]:
```suggestion
```

File: src/base_datamodule.py [hunk 4]
```diff
@@ -1,6 +1,14 @@
 from abc import ABC, abstractmethod
-from typing import Tuple, List
+from typing import List
 import numpy as np
+from dataclasses import dataclass
+
+@dataclass
+class NamedDataset:
+    name: str  
+    X: np.ndarray
```
Comment (noahho) [inline src/base_datamodule.py:9]: Don't we want to add a custom splitting function as well? Or a property such as is ordered or is grouped?

File: src/utils.py [hunk 5]
```diff
@@ -52,36 +47,31 @@ def train_and_predict_single_dataset(
 
 
 def benchmark_datasets(
-    model: xgb.XGBClassifier,
-    X_list: List[np.ndarray],
-    y_list: List[np.ndarray],
+    model: BaseEstimator,
+    dataset_list: List[NamedDataset],
     *,
     split_fn: Callable = lambda X, y: train_test_split(
```
Comment (noahho) [inline src/utils.py:53]: maybe the split function should be part of the dataset definition instead?

**Review by noahho — CHANGES_REQUESTED** [review]
Looks pretty good already. However, the splitting function might be better defined with the dataset / task. Also the metric that is optimized for might be defined somewhere else?

**Comment by noahho** [comment]
We should in the requirements guide ask for:
- metric they care about
- aggregation kind they care about
- estimated performance increase they care about
- datasets that are most important?
-

File: src/utils.py (hunk 5 above)
Comment (klemens-floege) [reply src/utils.py:53]: Yess, I agree

**Comment by noahho** [comment]
Please adress the below and then merge independelty, the rest looks good :-)

Only changes I'd propose were not made in this PR:
1. In README
```
    * **Related Tables:** Datasets should ideally be related, for example, sharing a similar domain or feature set.
### 3. Metadata (Nice to have):
* Please provide the following metadata for **each** set of datasets (train/test/validation lists):
    * `feature_names`: A list of strings corresponding to the columns in the `X` arrays (required if features are consistent).
```
to 
```
    * **Related Tables:** Datasets should be related, for example, sharing a similar domain or feature set.
### 3. Metadata (Optional):
* Please provide the following metadata for **each** set of datasets (train/test/validation lists):
    * `feature_names`: A list of strings corresponding to the columns in the `X` arrays (required if features are consistent).
    * `dataset_description`: A strong describing the content of each dataset
```
2. Let's merge the categories "Nice to have" and "Optional" to only "Optional".
3. How is all this data specified e.g. metadata and key metrics. Should users modify the README? Provide a description on how to submit this. Could be in README or some form they fill.

**Review by noahho — APPROVED** [review]
Please just look at my comments
